entities/schema/collection.go

```diff
@@ -0,0 +1,367 @@
+
+	// Can be a reference to another type when it starts with a capital (for example Person), otherwise "string" or "int".
+	// TODO-RAFT: Can we make DataType a slice of interface where other type and native type implements it ?
+	DataType []string `json:"data_type"`
+
+	// Optional. Should this property be indexed in the inverted index. Defaults to true. If you choose false, you will not be able to use this property in where filters. This property has no affect on vectorization decisions done by modules
+	IndexFilterable bool `json:"indexFilterable,omitempty"`
+
+	// Optional. Should this property be indexed in the inverted index. Defaults to true. If you choose false, you will not be able to use this property in where filters, bm25 or hybrid search. This property has no affect on vectorization decisions done by modules (deprecated as of v1.19; use indexFilterable or/and indexSearchable instead)
+	IndexInverted bool `json:"indexInverted,omitempty"`
+
+	// Optional. Should this property be indexed in the inverted index. Defaults to true. Applicable only to properties of data type text and text[]. If you choose false, you will not be able to use this property in bm25 or hybrid search. This property has no affect on vectorization decisions done by modules
+	IndexSearchable bool `json:"indexSearchable,omitempty"`
+
+	// Configuration specific to modules this Weaviate instance has installed
+	ModuleConfig map[string]interface{} `json:"moduleConfig,omitempty"`
+
+	// The properties of the nested object(s). Applies to object and object[] data types.
+	NestedProperties []NestedProperty `json:"nestedProperties,omitempty"`
+
+	// Determines tokenization of the property as separate words or whole field. Optional. Applies to text and text[] data types. Allowed values are `word` (default; splits on any non-alphanumerical, lowercases), `lowercase` (splits on white spaces, lowercases), `whitespace` (splits on white spaces), `field` (trims). Not supported for remaining data types
+	// Enum: [word lowercase whitespace field]
+	Tokenization string `json:"tokenization,omitempty"`
+}
+
+type NestedProperty struct {
+	// name
+	Name string `json:"name,omitempty"`
+	// description
+	Description string `json:"description,omitempty"`
+	// data type
+	DataType []string `json:"data_type"`
+
+	// index filterable
+	IndexFilterable bool `json:"index_filterable,omitempty"`
+
+	// index searchable
+	IndexSearchable bool `json:"index_searchable,omitempty"`
+
+	// nested properties
+	NestedProperties []NestedProperty `json:"nested_properties,omitempty"`
+
+	// tokenization
+	// Enum: [word lowercase whitespace field]
+	Tokenization string `json:"tokenization,omitempty"`
+}
+
+// NestedPropertyFromModel returns a NestedProperty copied from m.
+func NestedPropertyFromModel(m models.NestedProperty) NestedProperty {
+	n := NestedProperty{}
+
+	n.DataType = m.DataType
+	n.Description = m.Description
+	if m.IndexFilterable != nil {
+		n.IndexFilterable = *m.IndexFilterable
```

Comment:
This might be a good opportunity to remove all the boolean pointers.  There is no benefit to them being pointers, and it confuses everyone who sees them for the first time.

Expecially because nil defaults to true, which is not what most people expect.

Reply:
@redouan-rhazouani do we have a list of optional parameters ? Maybe a docs link I can use ? Otherwise I'll check the swagger config and deduce from there.

I'll change these parameters to defaults to true if no value.
Do we still want to have a pointer ?

Reply:
How do you mean revising?  I would like to see the pointers dissappear, and all booleans should be true or false.  We can apply the defaults when the class is loaded.

We don't make use of the optional part, I'd like to see that part removed.  It's especially confusing since programmers would assume nill implies false.

Reply:
@reyreaud-l 
Unfortunately, we don't have a list of optional parameters. The only way to determine the default value, is to inspect its usage in the code

Reply:
@donomii
The new Collection does not use pointers to boolean. The `Collection` is intended to replace `models.Class `throughout the codebase, except when dealing with swagger api. We are not changing `models.*` in this task, but rather, we aim to create a replacement that maintains backward compatibility

Reply:
Checking the swagger definition, the boolean indeed defaults to true. I pushed a change for that and updated the tests. I'll be checking manually in the code if we other "defaults" that are not golang defaults on other parameters.